### ISSUE TITLE
Couldn't parse `txMetaData` in wallet SQLite state machine tests

### ISSUE THREAD

**Issue by jonathanknowles** [cardano-foundation/cardano-wallet#2130]
# Context
 
https://hydra.iohk.io/build/4087307/nixlog/1

# Test Case

```
Cardano.Wallet.DB.Sqlite, Sqlite State machine tests, Sequential
```

# Failure / Counter-example

```hs

  test/unit/Cardano/Wallet/DB/SqliteSpec.hs:243:9: 
  1) Cardano.Wallet.DB.Sqlite, Sqlite State machine tests, Sequential
       uncaught exception: PersistException
       PersistMarshalError "Couldn't parse field `txMetaData` from table `tx_meta`. ConversionErrExpected \"[Aeson.String x]\" \"fromList [(\\\"hex\\\",Array [Number 0.0])]\""
       (after 1491 tests and 68 shrinks)
```
and
```hs
  test/unit/Cardano/Wallet/DB/SqliteSpec.hs:252:9: 
  2) Cardano.Wallet.DB.Sqlite, Sqlite State machine (RndState), Sequential state machine tests
       uncaught exception: PersistException
       PersistMarshalError "Couldn't parse field `txMetaData` from table `tx_meta`. ConversionErrExpected \"[Aeson.String x]\" \"fromList [(\\\"hex\\\",Array [Number 0.0])]\""
       (after 1491 tests and 70 shrinks)
```
# Resolution 

- The current metadata to/from json conversions is broken. We've [proposed an alternative](https://github.com/input-output-hk/cardano-node/pull/1797) that roundtrips for data coming from JSON. This is used in #2131 which fixes conversions issues. 

---

# QA

**Comment by jonathanknowles** [comment]
Potentially related: https://github.com/input-output-hk/cardano-node/pull/1797

**Comment by KtorZ** [comment]
Should be fixed by #2131
